In [95]:
import os
import pandas as pd
import numpy as np
from config import RAW_DATA_DIR, BASE_DATOS_DIR

In [97]:
df=pd.read_csv(RAW_DATA_DIR, index_col=False, low_memory=False, na_values=' ')
df

ENTIDAD  MUNICIPIO    CODIGO  ID_ESTRATO      UE        A111A  \
0            14        NaN       NaN         NaN  335120  1390382.820   
1            14        NaN       NaN         1.0  314288   189282.583   
2            14        NaN       NaN         2.0   16836   186386.076   
3            14        NaN       NaN         3.0    3212   327118.708   
4            14        NaN       NaN         4.0     784   687595.453   
...         ...        ...       ...         ...     ...          ...   
115584       14      125.0    8132.0        99.0       2          NaN   
115585       14      125.0   81323.0         NaN       2          NaN   
115586       14      125.0   81323.0        99.0       2          NaN   
115587       14      125.0  813230.0         NaN       2          NaN   
115588       14      125.0  813230.0        99.0       2          NaN   

             A121A       A131A      A211A      A221A  ...       A256A  \
0       795220.954  595161.866  30443.430  30646.122  ...  140420.765   
1        78003.045  111279.538   2755.271   3764.377  ...   22120.685   
2        86283.718  100102.358   2666.123   3263.430  ...  127318.483   
3       187622.233  139496.475   8755.796   8325.871  ...  160304.594   
4       443311.958  244283.495  16266.240  15292.444  ...  301420.045   
...            ...         ...        ...        ...  ...         ...   
115584         NaN         NaN        NaN        NaN  ...       0.000   
115585         NaN         NaN        NaN        NaN  ...       0.000   
115586         NaN         NaN        NaN        NaN  ...       0.000   
115587         NaN         NaN        NaN        NaN  ...       0.000   
115588         NaN         NaN        NaN        NaN  ...       0.000   

          A433A    A434A        A436A   A529A   A534A    A747A    A748A  \
0       174.666   95.637   977880.329   6.589  30.729  294.154  494.744   
1       172.369   97.214   451237.072   8.244  35.043  212.056  294.278   
2       154.706   96.234  1085990.597   5.480  25.774  223.673  350.472   
3       160.676   95.956  1543161.106   8.206  26.601  298.532  438.346   
4       203.243   94.329  1297820.791   6.069  34.621  316.962  526.604   
...         ...      ...          ...     ...     ...      ...      ...   
115584   91.549  100.000    21666.667  11.484 -78.758  195.602    0.000   
115585   91.549  100.000    21666.667  11.484 -78.758  195.602    0.000   
115586   91.549  100.000    21666.667  11.484 -78.758  195.602    0.000   
115587   91.549  100.000    21666.667  11.484 -78.758  195.602    0.000   
115588   91.549  100.000    21666.667  11.484 -78.758  195.602    0.000   

          A752A   A764A  
0        81.881  41.641  
1       108.227  48.139  
2        45.763  40.397  
3       105.337  34.835  
4        89.522  37.533  
...         ...     ...  
115584  -37.903  66.667  
115585  -37.903  66.667  
115586  -37.903  66.667  
115587  -37.903  66.667  
115588  -37.903  66.667  

[115589 rows x 186 columns]

In [98]:
# informacion de los datos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115589 entries, 0 to 115588
Columns: 186 entries, ENTIDAD to A764A
dtypes: float64(184), int64(2)
memory usage: 164.0 MB


Hay 115589 observaciones y 186 variables en la base de datos

In [99]:
# variables
for col in df.columns:
    print(f'La variable {col} tiene {df[col].nunique()} datos unicos y {df[col].isnull().sum()} datos perdidos')

La variable ENTIDAD tiene 1 datos unicos y 0 datos perdidos
La variable MUNICIPIO tiene 125 datos unicos y 4866 datos perdidos
La variable CODIGO tiene 1729 datos unicos y 359 datos perdidos
La variable ID_ESTRATO tiene 5 datos unicos y 55603 datos perdidos
La variable UE tiene 1387 datos unicos y 0 datos perdidos
La variable A111A tiene 16867 datos unicos y 60722 datos perdidos
La variable A121A tiene 14153 datos unicos y 60722 datos perdidos
La variable A131A tiene 15377 datos unicos y 60722 datos perdidos
La variable A211A tiene 7918 datos unicos y 60722 datos perdidos
La variable A221A tiene 6466 datos unicos y 60722 datos perdidos
La variable A511A tiene 7837 datos unicos y 60722 datos perdidos
La variable A700A tiene 16346 datos unicos y 60722 datos perdidos
La variable A800A tiene 17840 datos unicos y 60722 datos perdidos
La variable H000A tiene 3136 datos unicos y 60722 datos perdidos
La variable H000B tiene 2507 datos unicos y 60722 datos perdidos
La variable H000C tiene 2073 

In [100]:
# lista de los municipios en el dataset 
df['MUNICIPIO'].unique()

array([ nan,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,  10.,
        11.,  12.,  13.,  14.,  15.,  16.,  17.,  18.,  19.,  20.,  21.,
        22.,  23.,  24.,  25.,  26.,  27.,  28.,  29.,  30.,  31.,  32.,
        33.,  34.,  35.,  36.,  37.,  38.,  39.,  40.,  41.,  42.,  43.,
        44.,  45.,  46.,  47.,  48.,  49.,  50.,  51.,  52.,  53.,  54.,
        55.,  56.,  57.,  58.,  59.,  60.,  61.,  62.,  63.,  64.,  65.,
        66.,  67.,  68.,  69.,  70.,  71.,  72.,  73.,  74.,  75.,  76.,
        77.,  78.,  79.,  80.,  81.,  82.,  83.,  84.,  85.,  86.,  87.,
        88.,  89.,  90.,  91.,  92.,  93.,  94.,  95.,  96.,  97.,  98.,
        99., 100., 101., 102., 103., 104., 105., 106., 107., 108., 109.,
       110., 111., 112., 113., 114., 115., 116., 117., 118., 119., 120.,
       121., 122., 123., 124., 125.])

Se tiene datos de todos los 125 municipios de jalisco.

Los datos marcados como "np.NaN" son los datos agregados para todo el Estado.

In [101]:
# separar los datos de todo el estado de jalisco de sus municipos
jalisco_agg=df[df['MUNICIPIO'].isnull()]

print('Numero de observaciones para el estado de jalisco como agregados:, ',  jalisco_agg.shape[0])
jalisco_agg.head(3)

Numero de observaciones para el estado de jalisco como agregados:,  4866


ENTIDAD  MUNICIPIO  CODIGO  ID_ESTRATO      UE        A111A       A121A  \
0       14        NaN     NaN         NaN  335120  1390382.820  795220.954   
1       14        NaN     NaN         1.0  314288   189282.583   78003.045   
2       14        NaN     NaN         2.0   16836   186386.076   86283.718   

        A131A      A211A      A221A  ...       A256A    A433A   A434A  \
0  595161.866  30443.430  30646.122  ...  140420.765  174.666  95.637   
1  111279.538   2755.271   3764.377  ...   22120.685  172.369  97.214   
2  100102.358   2666.123   3263.430  ...  127318.483  154.706  96.234   

         A436A  A529A   A534A    A747A    A748A    A752A   A764A  
0   977880.329  6.589  30.729  294.154  494.744   81.881  41.641  
1   451237.072  8.244  35.043  212.056  294.278  108.227  48.139  
2  1085990.597  5.480  25.774  223.673  350.472   45.763  40.397  

[3 rows x 186 columns]

In [102]:
# codificar al estado de kalisco como el municipio 0
jalisco_agg['MUNICIPIO']=jalisco_agg['MUNICIPIO'].fillna(0)

C:\Users\ernes\AppData\Local\Temp\ipykernel_19360\2532493825.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  jalisco_agg['MUNICIPIO']=jalisco_agg['MUNICIPIO'].fillna(0)


In [103]:
muns=df[df['MUNICIPIO'].notnull()]

print('Numero de observaciones en todal para munucipios:, ',  muns.shape[0])
muns.head(3)

Numero de observaciones en todal para munucipios:,  110723


ENTIDAD  MUNICIPIO  CODIGO  ID_ESTRATO    UE    A111A    A121A    A131A  \
4866       14        1.0     NaN         NaN  1089  646.703  228.203  418.500   
4867       14        1.0     NaN         1.0  1065  280.360  109.860  170.500   
4868       14        1.0     NaN         2.0    21  111.728   46.666   65.062   

       A211A   A221A  ...      A256A    A433A   A434A       A436A   A529A  \
4866  35.773  39.221  ...  60611.795  178.834  97.837  364849.644   7.054   
4867  31.269  30.955  ...  59124.837  163.955  96.746  219831.816   5.681   
4868   0.230   3.992  ...  18910.284  130.140  98.658  796846.827  10.341   

       A534A    A747A    A748A    A752A   A764A  
4866  44.712  187.585  348.661  101.310  54.634  
4867  35.715  169.521  249.595   56.524  55.700  
4868  13.946  156.631  303.813  112.658  47.265  

[3 rows x 186 columns]

La variable H001A indica la cantidad de trabjadores empleados en el sector indicado por la variable CODIGO

In [104]:
vars_relevantes=['ENTIDAD', 'MUNICIPIO','CODIGO', 'ID_ESTRATO', 'H001A']

In [105]:
# concatenar los dos dataframes de nuevo para exportarlos
df_trabajo=pd.concat([jalisco_agg, muns], axis=0)
df_trabajo=df_trabajo.loc[:, vars_relevantes]

In [106]:
# formatear correctamente los datos
df_trabajo['MUNICIPIO']=df_trabajo['MUNICIPIO'].astype(int)
df_trabajo

ENTIDAD  MUNICIPIO    CODIGO  ID_ESTRATO      H001A
0            14          0       NaN         NaN  2004175.0
1            14          0       NaN         1.0   768539.0
2            14          0       NaN         2.0   346697.0
3            14          0       NaN         3.0   337722.0
4            14          0       NaN         4.0   551217.0
...         ...        ...       ...         ...        ...
115584       14        125    8132.0        99.0        NaN
115585       14        125   81323.0         NaN        NaN
115586       14        125   81323.0        99.0        NaN
115587       14        125  813230.0         NaN        NaN
115588       14        125  813230.0        99.0        NaN

[115589 rows x 5 columns]

In [110]:
df_trabajo.to_csv(BASE_DATOS_DIR, index=False, mode='w')